design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,115 @@
+
+## Motivation
+
+Renkulab.io provides the option to run interactive sessions of a project. Among other things the session
+contains a jupyter notebook.
+
+The motivation behind this RFC is to provide a simple renku CLI sub-command that basically allows the
+starting and stopping of these interactive sessions.
+
+## Design Detail
+
+The proposed sub-commands of `renku session`, namely to run interactive sessions. The local session heavily relies on
+[docker](https://www.docker.com/). Docker is one of the most popular container engines, that eases the development
+and deployment of applications, by off-loading the installation of the runtime environment dependencies of the
+application.
+
+In order to start a local interactive session, just as in case of renkulab.io, it is required to provide
+a `Dockerfile`. This is by default provided by `renku init` when creating a renku project, but if the user
+is using a custom project template it might not be present.
+
+### Proposed Commands
+
+#### renku session start
+
+A command to start an interactive session.
+
+By default, the `Dockerfile` present in the given project is going to be used for building the docker image
+that's going to be used for running the interactive session. In case the docker image is not available the
+user is going to be prompted whether the user wants to build the docker image, as in some cases it can be
+time consuming.
+
+By using the `--remote [endpoint]` flag an interactive session will be spawned on the specified endpoint. It requires
+that the user has authenticated herself previously with the `renku login` command and aquired the authentication
+token for the endpoint.	If the endpoint is not defined the command will try to assume the endpoint set in the project's
+configuration if available.
+
+##### Detailed Parameter Description
+
+```
+renku session start [OPTIONS]
+
+--image <image_name>	override the docker image to be used for the interactive session.
+--remote [endpoint]	start a remote interactive session using the JWT aquired for the
+			specific endpoint with renku login.
+--cpu <num of cpus>	Specify how much of the available CPU resources a container can use.
+--disk <disk space>	Specify the amount of disk space for the container.
+--gpu <device>		Specify the GPU allocated for the container.
+--memory <max memory>	The maximum amount of memory the container can use.
+
+```
+
+#### renku session stop
+
+`renku session stop my_session` simply stops the given interactive session.
+
+One can shut down all the currently running interactive sessions at once with the `renku session stop --all`.
+
+##### Detailed Parameter Description
+
+```
+renku session stop [OPTIONS] <name>
+
+<name>      Name of the session that the user wants to stop.
+
+--all       Stops all the running containers.
+
+```
+
+#### renku session list
+
+The command to list all the currently running local interactive sessions, with the detailed information regarding
```

Comment:
is this listing the sessions for this project or renku sessions in general? Should there be a corresponding `renku session open/connect` command that will automatically open the browser with the correct URL?

Reply:
the listing would only be only done for the `active project` (meaning based on the cwd). 
but yeah you are right i should add an example of the `renku session list`; currently this is my idea:

```
ID         STATUS    URL
---------  -------   -------
deadbeef   running   http://localhost:31337
deeeeeed   running   https://renkulab.io/projects/vigsterkr/renku-mls-demo/sessions/show/vigsterkr--renku-2dmls-2ddemo-7c289dbd
```

this in case of even a quite basic terminal URLs can be opened directly.

or you reckon, it would be still nice to have an explicit command for opening, like `renku session open/connect`?

Reply:
Yeah I wouldn't think of this except I really like (and often use) the `hub pr show` command from the github cli which opens up the current pr in the browser. Super useful to not have to copy/paste!

Reply:
hence my comment about "this in case of even a quite basic terminal URLs can be opened directly", see CMD+left-click in case of iTerm or simply just click in case of alacritty or CTRL+left-click in case of Gnome TERM

Reply:
but none the less adding such a comment like `renku session connect <ID>` shouldn't be too complicated. note as in case of `stop`, this sub-command does not require a `--remote` flag as it's unique as well.